app/src/main/java/edu/cascadia/mobas/pollinatorpathway/ui/login/UserDatabase.java

```diff
@@ -0,0 +1,27 @@
+package edu.cascadia.mobas.pollinatorpathway.ui.login;
+
+import android.content.Context;
+
+import androidx.room.Database;
+import androidx.room.Room;
+import androidx.room.RoomDatabase;
+
+@Database(entities = {Profile.class}, version = 1)
+public abstract class UserDatabase extends RoomDatabase {
+
+    private static final String dbName = "user";
+    private static UserDatabase userDatabase;
+
+    public static synchronized UserDatabase getUserDatabase(Context context){
+
+        if (userDatabase == null){
+            userDatabase = Room.databaseBuilder(context, UserDatabase.class, dbName)
+                    .fallbackToDestructiveMigration()
+                    .build();
+        }
+
+        return userDatabase;
+    }
```

Comment:
You will also need to run this method from the MainActivity. If you follow along with the book 1.3.6, you can call the method from the MainActivity's onCreate method, but you will need to allow the database to run on the main thread by adding `.allowMainThreadQueries()` on line 19 in the databaseBuilder execution chain.

Reply:
So if I understand it right, I have to replace the line 19 by .allowMainThreadQueries(), right?